MCB-Nucleo/Core/Src/freertos.c

```diff
@@ -0,0 +1,454 @@
+
+/* Private define ------------------------------------------------------------*/
+/* USER CODE BEGIN PD */
+/* USER CODE END PD */
+
+/* Private macro -------------------------------------------------------------*/
+/* USER CODE BEGIN PM */
+/* USER CODE END PM */
+
+/* Private variables ---------------------------------------------------------*/
+/* USER CODE BEGIN Variables */
+uint16_t ADC_throttle_val;
+uint16_t ADC_regen_val;
+
+
+
+struct InputFlags event_flags; // Event flags for deciding what state to be in.
+enum DriveState state;
+float cruise_velocity; // Velocity for cruise control
+float velocity_of_car; // Current velocity of the car will be stored here.
+uint8_t battery_soc; // Stores the charge of the battery, updated in a task.
+
+/*
+ *  These are global so I can view their value in the Live Expressions tab, will move to task later.
+ */
+float current;
+float velocity;
+
+/* USER CODE END Variables */
+/* Definitions for defaultTask */
+osThreadId_t defaultTaskHandle;
+const osThreadAttr_t defaultTask_attributes = {
+  .name = "defaultTask",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for GetADCValues */
+osThreadId_t GetADCValuesHandle;
+const osThreadAttr_t GetADCValues_attributes = {
+  .name = "GetADCValues",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for updateFlags */
+osThreadId_t updateFlagsHandle;
+const osThreadAttr_t updateFlags_attributes = {
+  .name = "updateFlags",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for motorStateMachi */
+osThreadId_t motorStateMachiHandle;
+const osThreadAttr_t motorStateMachi_attributes = {
+  .name = "motorStateMachi",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for getCANVelocity */
+osThreadId_t getCANVelocityHandle;
+const osThreadAttr_t getCANVelocity_attributes = {
+  .name = "getCANVelocity",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for getCANBatterySO */
+osThreadId_t getCANBatterySOHandle;
+const osThreadAttr_t getCANBatterySO_attributes = {
+  .name = "getCANBatterySO",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+/* Definitions for updateState */
+osThreadId_t updateStateHandle;
+const osThreadAttr_t updateState_attributes = {
+  .name = "updateState",
+  .stack_size = 128 * 4,
+  .priority = (osPriority_t) osPriorityNormal,
+};
+
+/* Private function prototypes -----------------------------------------------*/
+/* USER CODE BEGIN FunctionPrototypes */
+
+/* USER CODE END FunctionPrototypes */
+
+void StartDefaultTask(void *argument);
+void getADCValues(void *argument);
+void startUpdateFlags(void *argument);
+void startMotorStateMachine(void *argument);
+void startGetCANVelocity(void *argument);
+void StartSetCANBatterySO(void *argument);
+void StartUpdateState(void *argument);
```

Comment:
Are all/any of these threads "starting" something? I would question the naming here.

Reply:
FreeRTOS required both a "Task Name" as well as a "Entry Function." 
I used the same naming convention as the defaultTask that auto-generates with the program. 
("defaultTask" as the Task Name and "StartDefaultTask" as the Entry Fuction").
I agree that the use of "start" is not intuitive so I changed it to the same naming convention I use at work.

Reply:
Is this just a cube thing? Could they not be named the same thing? I don't see where the task name shows up in the code